quantecon/__init__.py

```diff
@@ -12,42 +12,51 @@
 		"Cannot import numba from current anaconda distribution. \
 			Please run `conda install numba` to install the latest version.")
 
-#-Modules-#
+# Modules
 from . import distributions
 from . import game_theory
 from . import quad
 from . import random
 from . import optimize
 
-#-Objects-#
-from .compute_fp import compute_fixed_point
-from .discrete_rv import DiscreteRV
-from .dle import DLE
-from .ecdf import ECDF
-from .estspec import smooth, periodogram, ar_periodogram
-# from .game_theory import <objects-here> 							#Place Holder if we wish to promote any general objects to the qe namespace.
-from .graph_tools import DiGraph, random_tournament_graph
-from .gridtools import (
-	cartesian, mlinspace, cartesian_nearest_index, simplex_grid, simplex_index
-)
-from .inequality import lorenz_curve, gini_coefficient, shorrocks_index, \
-	rank_size
-from .kalman import Kalman
-from .lae import LAE
-from .arma import ARMA
-from .lqcontrol import LQ, LQMarkov
-from .filter import hamilton_filter
-from .lqnash import nnash
-from .lss import LinearStateSpace
-from .matrix_eqn import solve_discrete_lyapunov, solve_discrete_riccati
-from .quadsums import var_quadratic_sum, m_quadratic_sum
-#->Propose Delete From Top Level
-#Promote to keep current examples working
+# Objects
+from ._compute_fp import compute_fixed_point
+from ._discrete_rv import DiscreteRV
+from ._dle import DLE
+from ._ecdf import ECDF
+from ._estspec import smooth, periodogram, ar_periodogram
+from ._graph_tools import DiGraph, random_tournament_graph
+from ._gridtools import (cartesian, mlinspace, simplex_grid, simplex_index,
+						 num_compositions)
+from ._inequality import (lorenz_curve, gini_coefficient, shorrocks_index,
+						  rank_size)
+from ._kalman import Kalman
+from ._lae import LAE
+from ._arma import ARMA
+from ._lqcontrol import LQ, LQMarkov
+from ._filter import hamilton_filter
+from ._lqnash import nnash
+from ._ivp import IVP
+from ._lss import LinearStateSpace
+from ._matrix_eqn import solve_discrete_lyapunov, solve_discrete_riccati
+from ._quadsums import var_quadratic_sum, m_quadratic_sum
+from ._rank_nullspace import rank_est, nullspace
+from ._robustlq import RBLQ
+
+#-> Propose Delete From Top Level
+# Promote to keep current examples working
 from .markov import MarkovChain, random_markov_chain, random_stochastic_matrix, \
 	gth_solve, tauchen, rouwenhorst
-#Imports that Should be Deprecated with markov package
-from .markov import mc_compute_stationary, mc_sample_path
-#<-
-from .rank_nullspace import rank_est, nullspace
-from .robustlq import RBLQ
+
+
 from .util import searchsorted, fetch_nb_dependencies, tic, tac, toc
+#<-
+
+# Imports that should be deprecated with markov package
+from .markov import mc_compute_stationary, mc_sample_path
+
+# Imports that are deprecated and will be removed in further versions
```

Comment:
@Smit-create do we have any infrastructure that issues a `Warning` for users that do import these that advises them to move towards the new import path (in version `0.7.0`) etc.?

Reply:
I don't want to hold up the release but if we don't then perhaps we can add this to help with migration given these are still available at the top level.

Reply:
This works in the following way:
```py
>>> from quantecon.ecdf import ECDF  # works but throws a warning
<stdin>:1: DeprecationWarning: Please use `ECDF` from the `quantecon` namespace, the `quantecon.ecdf` namespace is deprecated. You can use the following instead:
 `from quantecon import ECDF`.
>>> from quantecon import ECDF
```

Reply:
thanks @Smit-create I thought that was the case. Once we identify a deprecation point I might update these in a future release to let users know when that will happen.